tasks/bal_sorswap.ts

```diff
@@ -123,49 +127,68 @@ task('bal_sorswap', 'Swap 2 tokens via Balancer SOR', async (args, hre: HardhatR
 
   const networkERC20s = getNetworkERC20s(hre);
 
+  // If true(swaps erc20 for erc20) else (swaps eth for erc20)
+  const isErc20Swap = true;
+
   const queryOnChain = true;
   const swapType = SwapTypes.SwapExactIn;
   const ethDescriptor = networkERC20s[networkInfo]['ETH'];
   const usdcDescriptor = networkERC20s[networkInfo]['USDC'];
-  const swapAmount = new BigNumber(1);
-
-  const swapInfo = await getSwap(
-    provider,
-    networkInfo,
-    queryOnChain,
-    swapType,
-    ethDescriptor,
-    usdcDescriptor,
-    swapAmount,
-  );
-
-  console.log(`SwapInfo = ${JSON.stringify(swapInfo, undefined, 2)}`);
+  const balDescriptor = networkERC20s[networkInfo]['BAL'];
+  const swapAmount = new BigNumber(100);
 
+  let swapInfo: SwapInfo;
+  if (isErc20Swap) {
+    swapInfo = await getSwap(provider, networkInfo, queryOnChain, swapType, balDescriptor, usdcDescriptor, swapAmount);
+  } else {
+    swapInfo = await getSwap(provider, networkInfo, queryOnChain, swapType, ethDescriptor, usdcDescriptor, swapAmount);
+  }
   if (!swapInfo.returnAmount.gt(0)) {
     console.log(`Return amount is 0. No swaps to execute.`);
     return;
   }
 
-  if (swapInfo.tokenIn !== hre.ethers.constants.AddressZero) {
-    // TODO ERC20 allowance/approve
-    // // Vault needs approval for swapping non ETH
-    // console.log('Checking vault allowance...');
-    // const tokenInContract = new Contract(swapInfo.tokenIn, erc20abi, provider);
-    // let allowance = await tokenInContract.allowance(wallet.address, vaultAddr);
-    // if (bnum(allowance).lt(swapInfo.swapAmount)) {
-    //   console.log(`Not Enough Allowance: ${allowance.toString()}. Approving vault now...`);
-    //   const txApprove = await tokenInContract.connect(wallet).approve(vaultAddr, MaxUint256);
-    //   await txApprove.wait();
-    //   console.log(`Allowance updated: ${txApprove.hash}`);
-    //   allowance = await tokenInContract.allowance(wallet.address, vaultAddr);
-    // }
-    // console.log(`Allowance: ${allowance.toString()}`);
+  const ierc20abi = [
```

Comment:
as use of the IERC20 interface is expanding can we just import it? something like:

`import type { IERC20, ParaSwapV4Adapter } from '../typechain';`

Reply:
Makes sense.